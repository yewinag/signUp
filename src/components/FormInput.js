import React, { Component } from 'react'
import { Text, View, TextInput } from 'react-native'

const FormInput = (props) => (
    <TextInput {...props} />
)
export default FormInput;
