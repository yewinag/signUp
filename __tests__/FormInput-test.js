import 'react-native';
import React from 'react';
import FormInput from '../src/components/FormInput';

import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

it('test render FormInput correctly', () => {
  const tree = renderer.create(<FormInput />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('test FormInput props', ()=> {
    const onChangeEmail = jest.fn();       
    const wrapper = shallow(
        <FormInput
            type="email"
            placeholder="yewin@mail.com" 
            onChange={onChangeEmail}/>)            
    expect((wrapper.props().onChange)).toEqual(onChangeEmail)
    expect((wrapper.props().type)).toEqual("email")
    expect((wrapper.props().placeholder)).toEqual("yewin@mail.com")
})