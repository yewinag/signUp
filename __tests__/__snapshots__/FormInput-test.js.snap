// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render FormInput correctly 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
/>
`;
